src/signature.rs

```diff
@@ -134,8 +133,8 @@ pub fn verify(signature: &Signature, hashes: &[G2Projective], public_keys: &[Pub
         return false;
     }
 
-    // zero key & single hash should fail
-    if n_hashes == 1 && public_keys[0].0.is_identity().into() {
+    // zero keys should always fail.
+    if public_keys.iter().any(|pk| pk.0.is_identity().into()) {
```

Comment:
There's no reason to not check these up-front. We _could_ check them in parallel later, but we already have an `n^2` loop so this shouldn't have an appreciable impact on performance. Really, dropping the atomics will likely increase performance (even when taking relaxed ordering into account).